### DISCUSSION TITLE
Cypress Tests fail with TLS Error when using OWASP Zap as proxy server

### DISCUSSION THREAD

**Discussion by james-luther** [cypress-io/cypress#26961] [CI setup]
I'm attempting to use OWASP Zap as a proxy server to run all e2e tests through. When running all tests fail reporting the following:

We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

 Error: Client network socket disconnected before secure TLS connection was established

Common situations why this would fail:
- you don't have internet access
- you forgot to run / boot your web server
- your web server isn't accessible
- you have weird network configuration settings on your computer

Here is my cypress info:

❯ npx cypress info
Displaying Cypress info...

Detected 2 browsers installed:

1. Chrome
  - Name: chrome
  - Channel: stable
  - Version: 114.0.5735.106
  - Executable: /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
  - Profile: /Users/james/Library/Application Support/Cypress/cy/production/browsers/chrome-stable

2. Firefox
  - Name: firefox
  - Channel: stable
  - Version: 114.0
  - Executable: /Applications/Firefox.app/Contents/MacOS/firefox
  - Profile: /Users/james/Library/Application Support/Cypress/cy/production/browsers/firefox-stable

Note: to run these browsers, pass <name>:<channel> to the '--browser' field

Examples:
- cypress run --browser chrome
- cypress run --browser firefox

Learn More: https://on.cypress.io/launching-browsers

Proxy Settings:
HTTP_PROXY: http://192.168.3.7:8080
HTTPS_PROXY: https://192.168.3.7:8080
NO_PROXY: localhost

Learn More: https://on.cypress.io/proxy-configuration

Environment Variables: none detected

Application Data: /Users/james/Library/Application Support/cypress/cy/development
Browser Profiles: /Users/james/Library/Application Support/cypress/cy/development/browsers
Binary Caches: /Users/james/Library/Caches/Cypress

Cypress Version: 10.11.0 (stable)
System Platform: darwin (22.5.0)
System Memory: 17.2 GB free 1.22 GB

The ZAP proxy is working as I am able to manually reach and test the endpoints Cypress is testing. When I remove the proxy all tests complete successfully. I'm not sure where I am going wrong.

**Comment by 0xSk3ttch** [comment]
@james-luther Did you every determine the root cause for this? Running into it now and came across your post. Thanks!

**Reply by james-luther** [reply]
I did actually! I set the HTTPS proxy to the same endpoint as HTTP. I reached out to the team for ZAP and was informed they were working on this kind of use case but I never followed up after finding my solution.

**Reply by 0xSk3ttch** [reply]
@james-luther Thanks for the response! I've got those set, but still seeing it. Dang. Were / are you running in Docker?

**Reply by james-luther** [reply]
@jacobbundren I was, it was specifically running in CircleCI. The base container was ZAP and I installed everything for cypress. They both ran together

**Reply by 0xSk3ttch** [reply]
@james-luther I'm gong to give that a whirl. I've been using the Cypress base image. I'll report back. Thanks again for the insight.
